### DISCUSSION TITLE
Pie chart styles

### DISCUSSION THREAD

**Discussion by sosmsnov** [recharts/recharts#4654] [Q&A]
Hi.
I sorry for opening issue here, but I did not find way to communicate with you. 
Is it possible to make a Pie chart like that with Rechart? 
1. I could not make wisth on parts different
2. border radius
3. shadows
4. labels on parts
5. label inside the chart
6. dashed border inside the Pie
Expected:
<img width="426" alt="Screenshot 2024-06-11 at 3 47 03 PM" src="https://github.com/recharts/recharts/assets/160776375/56d7760d-1b42-45ed-883a-4e29fa40f947">

My result: 
<img width="297" alt="Screenshot 2024-06-11 at 3 51 32 PM" src="https://github.com/recharts/recharts/assets/160776375/ce27cff6-a56d-4f54-9fbf-3b3298499541">

**Comment by ckifer** [comment]
1. you can use `activeShape` and `activeIndex` for your active state
2. `Sector` supports a `cornerRadius` prop, but you currently can't specify it only for the outside corners. You could make a custom sector implementation and provide it to `inactiveShape` if you want with any svg components 
3. Not sure what you mean, I don't see any shadows in the mockups. But SVGs require you to use `filter` so `style={{ filter: "drop-shadow( 3px 3px 2px rgba(0, 0, 0, .7))" }}` should work
4. I see you have that already, you can do this using `LabelList`
5. I see you have that already, you can do this using `Label`
6. There is no way to do this at this time, you could create this yourself using svgs

**Comment by ckifer** [comment]
I just hacked this together so forgive how bad it is
https://codesandbox.io/p/sandbox/pie-chart-with-customized-active-shape-forked-jy4kjm?file=%2Fsrc%2FApp.tsx%3A41%2C29

**Comment by sosmsnov** [comment]
@ckifer  I'll try to use your suggestions. thank you a lot

**Comment by ckifer** [comment]
Going to convert this to a discussion since its a question

**Reply by vtbt** [reply]
Hi @ckifer 
About this point: 

> 1. you can use activeShape and activeIndex for your active state

Now, it's already recharts v3.x, and `activeIndex` was removed, so any udpate or solution for this discussion?
I got issue when upgrade from recharts v2.x from recharts v3.x, and  `activeIndex` is not existed, so dont know how to fix
Thank you very much for helping me.
